spyder/widgets/findinfiles.py

```diff
@@ -918,13 +932,15 @@ class FindInFilesWidget(QWidget):
     sig_finished = Signal()
 
     def __init__(self, parent,
-                 search_text=r"# ?TODO|# ?FIXME|# ?XXX",
-                 search_text_regexp=True,
-                 exclude=r"\.pyc$|\.orig$|\.hg|\.svn", exclude_idx=None,
-                 exclude_regexp=True,
+                 search_text="",
+                 search_text_regexp=False,
+                 exclude="*.json, *.csv, *.js",

```

Comment:
Is there a motivation for js as one of the default excludes (but not, e.g., HTML or CSS)?

Reply:
No problem. I'll add html and css.

What do you think about excluding json and csv by default? Perhaps it would be better to skip `*.dat` and `*.log` instead?

Reply:
> No problem. I'll add html and css.

I guess my comment was unclear; the intent of question was more "What's the motivation to add ``.js`` as default ignores in the first place" with HTML and CSS as examples of the resulting "slippery slope". *Personally,* I typically have ``,js`` in there myself as an inclusion, as I don't usually want to search the Mathjax files, jQuery, etc. in my searches of Spyder, Javascript stuff searching the docs or the website, etc., but that's all for particular cases and more oriented toward software or web development vs. data science that Spyder is targeted at so I'm not sure it really makes sense in the general case.

CSV certainly seems sensible to have, as it is one of the most comment, if not *the* most common plain-text data formats, which would often be very large and burdensome to search, while finding results in them are rarely desired. By the same token, ``.dat`` (which to my knowledge is used for other types of ASCII as well as binary data files) would probably similarly make sense. I'm not sure how much JSON is used in data science especially for large data storage, so while it probably doesn't hurt to include it, it might not be priority. Excluding temporary/ephemeral files is generally a good idea, so under that justification ``.log`` makes sense; I typically also include ``.bak`` and (Git) ``.orig`` for that purpose, and you could include ``.tmp`` as well. So, in summary, my suggestion would be something like ``*.bak, *.tmp, *.orig, *.log, *.csv, *.dat``

Reply:
Good suggestion. I'll change the default pattern for that one :+1: